.github/workflows/save_cache.yml

```diff
@@ -15,12 +15,20 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.RENKU_DOCKER_USERNAME }}
+          password: ${{ secrets.RENKU_DOCKER_PASSWORD }}
       - name: Install python deps
         uses: devcontainers/ci@v0.3
         with:
           runCmd: poetry install --with dev
-          push: never
+          push: always
           skipContainerUserIdUpdate: false
+          imageName: renku/renku-data-service
+          imageTag: devcontainer-main
+          cacheFrom: renku/renku-data-service:devcontainer-main
```

Comment:
I think using GHCR for this would be much faster (near storage in the same infrastructure vs remote registry).